byterun/floats.c

```diff
@@ -530,6 +650,49 @@ CAMLprim value caml_log1p_float(value f)
   return caml_copy_double(caml_log1p(Double_val(f)));
 }
 
+#ifndef HAS_C99_FLOAT_OPS
+static inline double simple_erf(double x)
+{
+  double a1 =  0.254829592, a2 = -0.284496736, a3 = 1.421413741,
+         a4 = -1.453152027, a5 =  1.061405429, p  = 0.3275911,
+         t, y;
+
+  int sign = (x >= 0) ? 1 : -1;
+  x = fabs(x);
+  t = 1.0 / (1.0 + p * x);
+  y = 1.0 - (((((a5 *t  + a4) * t) + a3) * t + a2) * t + a1) * t * exp(-x * x);
```

Comment:
I don't know what platform you are using, but I haven't run into practical examples yet where floats significantly outperformed doubles on modern CPUs.  They usually perform the same, but I have seen cases where doubles performed better.  I've just tried it out, and there is no appreciable difference in performance between `exp` and `expf` on my platform.

There are surely cases where floats might be a better choice, because you can fit more of them into the cache or might benefit from better vectorization, but I haven't run into anything like that myself.  There are numerous online discussions about this topic and the general advice is to use doubles unless you have very good reasons not to.  They are almost always at least as fast, sometimes faster, and obviously offer more precision.